cloud/services/agentpools/agentpools.go

```diff
@@ -87,20 +87,23 @@ func (s *Service) Reconcile(ctx context.Context, spec interface{}) error {
 		// Normalize individual agent pools to diff in case we need to update
 		existingProfile := containerservice.AgentPool{
 			ManagedClusterAgentPoolProfileProperties: &containerservice.ManagedClusterAgentPoolProfileProperties{
-				VMSize:              existingPool.ManagedClusterAgentPoolProfileProperties.VMSize,
-				OsType:              containerservice.Linux,
-				OsDiskSizeGB:        existingPool.ManagedClusterAgentPoolProfileProperties.OsDiskSizeGB,
 				Count:               existingPool.ManagedClusterAgentPoolProfileProperties.Count,
-				Type:                containerservice.VirtualMachineScaleSets,
 				OrchestratorVersion: existingPool.ManagedClusterAgentPoolProfileProperties.OrchestratorVersion,
-				VnetSubnetID:        existingPool.ManagedClusterAgentPoolProfileProperties.VnetSubnetID,
+			},
+		}
+
+		normalizedProfile := containerservice.AgentPool{
```

Comment:
what is the benefit of this change using normalizedProfile instead of the complete profile to compare?

Reply:
We make a check for only fields which are actually mutable.
Apart from count and Os version nothing is mutable in AKS agentpools. So normalizing it would prevent incorrect update to the azure agentpool service.

Reply:
the scale-up issue is because of 106 being missed out. i.e if a mode is not specified it takes its default value which is User. So when an update was triggered for scaling up  the mode was changed to User from System.

Reply:
> We make a check for only fields which are actually mutable.
> Apart from count and Os version nothing is mutable in AKS agentpools. So normalizing it would prevent incorrect update to the azure agentpool service.

but the issue is, the input to the CreateUpdate is Profile which takes info from spec, not existing profile. So it can't prevent the incorrect update ?